radial_build_tools.py

```diff
@@ -56,6 +54,38 @@ def __init__(self, build, **kwargs):
         ):
             self.__setattr__(name, kwargs[name])
 
+        self.used_colors = set()
+        self.available_colors = set(matplotlib.colors.XKCD_COLORS.values())
+        self.colors = self.assign_colors()
+
+    def assign_colors(self):
+        """
+        Assign colors to each layer. If a color is provided, use it.
+        If not, generate a random color that hasn't been used yet.
+        """
+        colors = []
+        for layer in self.build.values():
+            if "color" in layer:
+                color = layer["color"]
+                self.used_colors.add(color)
+                self.available_colors.discard(color)
+            else:
+                color = self.generate_unique_color()
+                layer["color"] = color  
```

Comment:
Your method to generate a unique color is one line plus some of the adding/removing colors from sets that you did here already for other cases.  This suggestion streamlines things a lot.
```suggestion
            if "color" not in layer:
                layer["color"] = random.choice(list(self.available_colors))
            color = layer["color"]
            self.used_colors.add(color)
            self.available_colors.discard(color)
```